docfx_yaml/extension.py

```diff
@@ -1417,58 +1417,127 @@ def find_markdown_pages(app, outdir):
                 'href': mdfile.name.lower(),
             })
 
+def resolve_cross_reference(
+    words: List[str],
+    index: int,
+    current_word: str,
+    keyword: str,
+    visited_words: List[str],
+    hard_coded_references: Dict[str, str] = None
+) -> Optional[str]:
+    """Given a word, check if it contains a reference keyword we should convert.
 
-# Finds and replaces occurrences which should be a cross reference in the given
-# content, except for the current name.
-def convert_cross_references(content: str, current_name: str, entry_names: List[str]):
+    Args:
+        words: list of words used to check and compare content before and after `current_word`
+        index: index position of `current_word` within words
+        current_word: current word being looked at
```

Comment:
Total nit: perhaps `current_word` should be the first arg?

Reply:
Done.